packages/nx/src/plugins/js/project-graph/build-dependencies/target-project-locator.ts

```diff
@@ -364,7 +369,11 @@ export class TargetProjectLocator {
         packageJsonPath ?? resolveRelativeToDir(packageName, relativeToDir);
       let dir = dirname(pathOfFileInPackage);
 
-      while (dir !== parse(dir).root) {
+      while (
+        // If we're at the root, then parse(dir).root is empty (''), so handle it separately.
+        dir !== '.' &&
+        dir !== parse(dir).root
```

Comment:
Can we get rid of this `parse` call.. let's use `dirname`.. 

`dir === dirname(dir)` will be true when it's reached the root. That's what we use everywhere else.